File: api/getIPInfo.py
import json
import requests

def handler(request):
    # Only allow GET requests
    if request.method != 'GET':
        return {
            'statusCode': 405,
            'body': json.dumps({'message': 'Method not allowed'}),
            'headers': {
                'Access-Control-Allow-Origin': '*',  # Allow all origins
                'Content-Type': 'application/json'
            }
        }

    # Get IP from query parameters
    ip = request.args.get('ip')
    api_token = 'f0b8fb50fc925f'  # Replace with your actual token
    APIEndpoint = f'https://ipinfo.io/{ip}?token={api_token}'

    try:
        response = requests.get(APIEndpoint)
        response.raise_for_status()  # Raise an error for bad responses
        data = response.json()  # Parse JSON data
        return {
            'statusCode': 200,
            'body': json.dumps(data),  # Return data as JSON
            'headers': {
                'Access-Control-Allow-Origin': '*',  # Allow all origins
                'Content-Type': 'application/json'
            }
        }
    except requests.RequestException as error:
        print('Error:', error)
        return {
            'statusCode': 500,
            'body': json.dumps({'message': 'Internal server error'}),
            'headers': {
                'Access-Control-Allow-Origin': '*',  # Allow all origins
                'Content-Type': 'application/json'
            }
        }

File: api/getIPInfo.js
// api/getIPInfo.js

const axios = require('axios');

export default async function handler(req, res) {
    // Handle CORS preflight requests
    if (req.method === 'OPTIONS') {
        res.setHeader('Access-Control-Allow-Origin', '*');
        res.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS');
        res.setHeader('Access-Control-Allow-Headers', 'Content-Type');
        res.status(204).end();
        return;
    }

    // Check if the request method is GET
    if (req.method !== 'GET') {
        res.setHeader('Access-Control-Allow-Origin', '*');
        return res.status(405).json({ message: 'Method not allowed' });
    }

    // Get the IP from query parameters
    const ip = req.query.ip;
    if (!ip) {
        res.setHeader('Access-Control-Allow-Origin', '*');
        return res.status(400).json({ message: 'IP parameter is required' });
    }

    const apiToken = 'f0b8fb50fc925f'; // Replace with your actual token
    const APIEndpoint = `https://ipinfo.io/${ip}?token=${apiToken}`;

    try {
        const response = await axios.get(APIEndpoint);
        res.setHeader('Access-Control-Allow-Origin', '*');
        return res.status(200).json(response.data);
    } catch (error) {
        res.setHeader('Access-Control-Allow-Origin', '*');
        return res.status(500).json({
            message: 'Internal server error',
            error: error.message || 'Unknown error',
        });
    }
}
